### PR TITLE
Upgrade apollo-server-koa

### PR DESCRIPTION
**Pull request by IkeLutra**
#### Description of what you did:

Updates `apollo-server-koa` to a version that fixes the security advisory in #7085

### CODE REVIEW

**Comment by derrickmehaffy** [comment]
I've forcefully retriggered the TravisCI PG test that failed.

File: packages/strapi-plugin-graphql/package.json [hunk 1]
```diff
@@ -12,7 +12,7 @@
   },
   "dependencies": {
     "@apollo/federation": "^0.15.0",
-    "apollo-server-koa": "2.9.12",
+    "apollo-server-koa": "^2.14.2",
```
Comment (alexandrebodin) [inline packages/strapi-plugin-graphql/package.json:15]: Please make use of strict versions :)

Comment (IkeLutra) [reply packages/strapi-plugin-graphql/package.json:15]: You mean like `2.14.2`? I'm happy to chnage but can you explain the reasoning for strict versioning on just this dependency?

Comment (alexandrebodin) [reply packages/strapi-plugin-graphql/package.json:15]: @IkeLutra it is not specific to this package but we are taking every occasion to move to strict versions :) We have a PR to add dependabot PRs so we can get automated upgrades for thise kind of issues and still have stable versions so we don't break in unintended ways when dependecies break (which happened a few times already) Hope this helps

Comment (IkeLutra) [reply packages/strapi-plugin-graphql/package.json:15]: Thanks that is really helpful. Just wanted to understand the reasoning that makes a lot of sense. Will update now 😄

**Pull request merged**